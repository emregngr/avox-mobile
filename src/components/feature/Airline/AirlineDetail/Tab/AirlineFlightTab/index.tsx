import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Alliance } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/Alliance'
import { DestinationList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/DestinationList'
import { RoutesList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface AirlineFlightTabProps {
  airlineData: Airline
}

export const AirlineFlightTab = ({ airlineData }: AirlineFlightTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { network, operations } = airlineData ?? {}
  const { alliance } = operations ?? {}
  const {
    destinationCount,
    destinationCountries,
    destinations,
    domesticConnections,
    internationalConnections,
    routes,
  } = network ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const iconColor = useMemo(() => colors.onPrimary100, [colors.onPrimary100])

  const localeStrings = useMemo(
    () => ({
      activeRoutes: getLocale('activeRoutes'),
      alliance: getLocale('alliance'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      mainDestinations: getLocale('mainDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={iconColor}
        stats={memoizedStats}
      />

      <DestinationList
        destinations={destinations}
        iconColor={iconColor}
        title={localeStrings.mainDestinations}
      />

      <RoutesList iconColor={iconColor} routes={routes} title={localeStrings.activeRoutes} />

      {alliance !== 'none' && (
        <Alliance alliance={alliance} iconColor={iconColor} title={localeStrings.alliance} />
      )}
    </View>
  )
}
