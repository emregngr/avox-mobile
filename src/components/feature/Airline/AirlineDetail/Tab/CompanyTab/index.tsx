import React from 'react'
import { View } from 'react-native'

import { Company } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company'
import { Contact } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Contact'
import { Hub } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Hub'
import { Map } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Map'
import { SocialMedia } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/SocialMedia'
import type { Airline } from '@/types/feature/airline'

interface CompanyTabProps {
  airlineData: Airline
}

export const CompanyTab = ({ airlineData }: CompanyTabProps) => {
  const { companyInfo, operations } = airlineData

  return (
    <View className="px-4 gap-y-4">
      <Company airlineData={airlineData} />

      <Contact companyInfo={companyInfo} operations={operations} />

      <SocialMedia companyInfo={companyInfo} />

      <Hub operations={operations} />

      <Map airlineData={airlineData} />
    </View>
  )
}
