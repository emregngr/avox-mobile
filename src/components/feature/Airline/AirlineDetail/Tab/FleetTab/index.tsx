import { router } from 'expo-router'
import React, { useCallback, useMemo } from 'react'
import { View } from 'react-native'

import { FleetHeader } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetHeader'
import { FleetList } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetList'
import { FleetStats } from '@/components/feature/Airline/AirlineDetail/Tab/FleetTab/Sections/FleetStats'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface FleetTabProps {
  airlineData: Airline
}

export const FleetTab = ({ airlineData }: FleetTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { fleet, operations } = airlineData || {}

  const { region } = operations || {}

  const { airplaneTypeCount, airplanes, averageAgeYears, totalAirplane } = fleet || {}

  const sortedAirplanes = useMemo(
    () => airplanes?.sort((a, b) => b.count - a.count) || [],
    [airplanes],
  )

  const regionLower = useMemo(() => region?.toLowerCase(), [region])

  const handleImagePress = useCallback(
    (airplaneType: string, imageKey: string) => {
      router.navigate({
        params: {
          regionLower,
          selectedImageKey: imageKey,
          title: airplaneType,
        },
        pathname: '/image-modal',
      })
    },
    [regionLower],
  )

  const localeStrings = useMemo(
    () => ({
      fleetDetail: getLocale('fleetDetail'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <FleetStats
        airplaneTypeCount={airplaneTypeCount}
        averageAgeYears={averageAgeYears}
        totalAirplane={totalAirplane}
      />

      <FleetHeader colors={colors} fleetDetailText={localeStrings.fleetDetail} />

      <FleetList
        airplanes={sortedAirplanes}
        colors={colors}
        onImagePress={handleImagePress}
        region={region}
        totalAirplane={totalAirplane}
      />
    </View>
  )
}
