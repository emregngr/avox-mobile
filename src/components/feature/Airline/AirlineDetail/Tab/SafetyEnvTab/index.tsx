import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Environmental } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Sections/Environmental'
import { Safety } from '@/components/feature/Airline/AirlineDetail/Tab/SafetyEnvTab/Sections/Safety'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirlineType } from '@/types/feature/airline'

interface SafetyEnvTabProps {
  airlineData: AirlineType
}

export const SafetyEnvTab = ({ airlineData }: SafetyEnvTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { environmental, safety } = airlineData ?? {}
  const { certifications, safetyRecord } = safety ?? {}

  const localeStrings = useMemo(
    () => ({
      certifications: getLocale('certifications'),
      environmentalPolicy: getLocale('environmentalPolicy'),
      environmentalResponsibility: getLocale('environmentalResponsibility'),
      safetyInformation: getLocale('safetyInformation'),
      safetyRecord: getLocale('safetyRecord'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <Safety
        certifications={certifications}
        certificationsTitle={localeStrings.certifications}
        iconColor={colors.onPrimary100}
        safetyRecord={safetyRecord}
        safetyRecordTitle={localeStrings.safetyRecord}
        title={localeStrings.safetyInformation}
      />

      <Environmental
        content={environmental}
        iconColor={colors.onPrimary100}
        subtitle={localeStrings.environmentalResponsibility}
        title={localeStrings.environmentalPolicy}
      />
    </View>
  )
}
