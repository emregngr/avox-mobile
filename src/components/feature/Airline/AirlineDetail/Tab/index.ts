export * from './AirlineFlightTab'
export * from './CompanyTab'
export * from './FleetTab'
export * from './SafetyEnvTab'
