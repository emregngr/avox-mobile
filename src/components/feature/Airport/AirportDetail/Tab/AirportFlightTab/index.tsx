import { useMemo } from 'react'
import { View } from 'react-native'

import { AirlinesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/AirlinesList'
import { RoutesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface AirportFlightTabProps {
  airportData: AirportType
}

export const AirportFlightTab = ({ airportData }: AirportFlightTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { flightOperations } = airportData ?? {}
  const {
    airlines,
    destinationCount,
    destinationCountries,
    domesticConnections,
    internationalConnections,
    routes,
  } = flightOperations ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const localeStrings = useMemo(
    () => ({
      airlines: getLocale('airlines'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      popularDestinations: getLocale('popularDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={colors?.onPrimary100}
        stats={memoizedStats}
      />

      <AirlinesList
        airlines={airlines}
        iconColor={colors?.onPrimary100}
        title={localeStrings.airlines}
      />

      <RoutesList
        iconColor={colors?.onPrimary100}
        routes={routes}
        title={localeStrings.popularDestinations}
      />
    </View>
  )
}
