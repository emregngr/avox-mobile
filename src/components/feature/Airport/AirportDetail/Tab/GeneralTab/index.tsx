import React from 'react'
import { View } from 'react-native'

import { Contact } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Contact'
import { General } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/General'
import { Location } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Location'
import { Map } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map'
import { Services } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Services'
import { SocialMedia } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/SocialMedia'
import type { Airport } from '@/types/feature/airport'

interface GeneralTabProps {
  airportData: Airport
}

export const GeneralTab = ({ airportData }: GeneralTabProps) => {
  const { airportInfo, facilities, operations } = airportData ?? {}

  return (
    <View className="px-4 gap-y-4">
      <General airportInfo={airportInfo} operations={operations} />

      <Services facilities={facilities} />

      <Contact airportInfo={airportInfo} />

      <SocialMedia airportInfo={airportInfo} />

      <Location operations={operations} />

      <Map airportData={airportData} />
    </View>
  )
}
