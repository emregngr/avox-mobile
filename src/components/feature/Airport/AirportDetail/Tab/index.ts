export * from './AirportFlightTab'
export * from './GeneralTab'
export * from './InfrastructureTab'
export * from './NearbyPlacesTab'
