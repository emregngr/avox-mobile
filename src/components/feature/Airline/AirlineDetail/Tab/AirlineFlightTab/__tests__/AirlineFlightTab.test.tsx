import { render } from '@testing-library/react-native'
import React from 'react'

import { AirlineFlightTab } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'

jest.mock('@/locales/i18next')

const mockedUseGetLocale = getLocale as jest.MockedFunction<typeof getLocale>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/store/theme')

const mockedUseThemeStore = useThemeStore as jest.MockedFunction<typeof useThemeStore>

const mockedStatsGrid = jest.fn()
const mockedDestinationList = jest.fn()
const mockedRoutesList = jest.fn()
const mockedAlliance = jest.fn()

jest.mock(
  '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/Alliance',
  () => {
    const { View, Text } = require('react-native')
    return {
      Alliance: (props: { alliance: string; title: string }) => {
        mockedAlliance(props)
        if (!props.alliance || props.alliance === 'none') return null
        return (
          <View testID="alliance-section">
            <Text testID="alliance-title">{props.title}</Text>
            <Text testID="alliance-value">{props.alliance}</Text>
          </View>
        )
      },
    }
  },
)

jest.mock(
  '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/DestinationList',
  () => {
    const { View, Text } = require('react-native')
    return {
      DestinationList: (props: { destinations: any; title: string }) => {
        mockedDestinationList(props)
        return (
          <View testID="destination-list-section">
            <Text testID="destination-title">{props.title}</Text>
            <Text testID="destination-count">{props.destinations?.length || 0}</Text>
          </View>
        )
      },
    }
  },
)

jest.mock(
  '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/RoutesList',
  () => {
    const { View, Text } = require('react-native')
    return {
      RoutesList: (props: { routes: any; title: string }) => {
        mockedRoutesList(props)
        return (
          <View testID="routes-list-section">
            <Text testID="routes-title">{props.title}</Text>
            <Text testID="routes-count">{props.routes?.length || 0}</Text>
          </View>
        )
      },
    }
  },
)

jest.mock(
  '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/StatsGrid',
  () => {
    const { View, Text } = require('react-native')
    return {
      StatsGrid: (props: { localeStrings: any; stats: any }) => {
        mockedStatsGrid(props)
        return (
          <View testID="stats-grid-section">
            <Text testID="destination-count-stat">{props.stats.destinationCount || 0}</Text>
            <Text testID="destination-countries-stat">{props.stats.destinationCountries || 0}</Text>
            <Text testID="domestic-connections-stat">{props.stats.domesticConnections || 0}</Text>
            <Text testID="international-connections-stat">
              {props.stats.internationalConnections || 0}
            </Text>
            <Text testID="country">{props.localeStrings.country}</Text>
            <Text testID="domestic">{props.localeStrings.domesticDestinations}</Text>
            <Text testID="international">{props.localeStrings.internationalDestination}</Text>
            <Text testID="total">{props.localeStrings.totalDestination}</Text>
          </View>
        )
      },
    }
  },
)

const mockedAirlineData: any = {
  network: {
    destinationCount: 150,
    destinationCountries: 45,
    destinations: [
      { code: 'JFK', name: 'John F. Kennedy International Airport', city: 'New York' },
      { code: 'LAX', name: 'Los Angeles International Airport', city: 'Los Angeles' },
      { code: 'LHR', name: 'London Heathrow Airport', city: 'London' },
    ],
    domesticConnections: 80,
    internationalConnections: 70,
    routes: [
      { from: 'JFK', to: 'LAX', frequency: 'Daily' },
      { from: 'JFK', to: 'LHR', frequency: '3x weekly' },
    ],
  },
  operations: {
    alliance: 'Star Alliance',
  },
}

const mockedAirlineDataWithoutAlliance: any = {
  ...mockedAirlineData,
  operations: {
    alliance: 'none',
  },
}

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({ selectedLocale: 'en' })
  mockedUseThemeStore.mockReturnValue({ selectedTheme: 'light' })

  mockedUseGetLocale.mockImplementation((key: string) => key)
})

describe('AirlineFlightTab Component', () => {
  describe('Component Rendering', () => {
    it('renders all main sections when data is provided', () => {
      const { getByTestId } = render(<AirlineFlightTab airlineData={mockedAirlineData} />)
      expect(getByTestId('stats-grid-section')).toBeTruthy()
      expect(getByTestId('destination-list-section')).toBeTruthy()
      expect(getByTestId('routes-list-section')).toBeTruthy()
      expect(getByTestId('alliance-section')).toBeTruthy()
    })

    it('does not render Alliance section when alliance is "none"', () => {
      const { getByTestId, queryByTestId } = render(
        <AirlineFlightTab airlineData={mockedAirlineDataWithoutAlliance} />,
      )
      expect(getByTestId('stats-grid-section')).toBeTruthy()
      expect(getByTestId('destination-list-section')).toBeTruthy()
      expect(getByTestId('routes-list-section')).toBeTruthy()
      expect(queryByTestId('alliance-section')).toBeNull()
    })
  })

  describe('Prop Passing Checks', () => {
    it('should pass correct props to all child components', () => {
      render(<AirlineFlightTab airlineData={mockedAirlineData} />)
      expect(mockedStatsGrid).toHaveBeenCalledWith(
        expect.objectContaining({
          stats: {
            destinationCount: mockedAirlineData.network.destinationCount,
            destinationCountries: mockedAirlineData.network.destinationCountries,
            domesticConnections: mockedAirlineData.network.domesticConnections,
            internationalConnections: mockedAirlineData.network.internationalConnections,
          },
        }),
      )
      expect(mockedDestinationList).toHaveBeenCalledWith(
        expect.objectContaining({
          destinations: mockedAirlineData.network.destinations,
        }),
      )
      expect(mockedRoutesList).toHaveBeenCalledWith(
        expect.objectContaining({
          routes: mockedAirlineData.network.routes,
        }),
      )
      expect(mockedAlliance).toHaveBeenCalledWith(
        expect.objectContaining({
          alliance: mockedAirlineData.operations.alliance,
        }),
      )
    })

    it('should call all components exactly once when data is full', () => {
      render(<AirlineFlightTab airlineData={mockedAirlineData} />)
      expect(mockedStatsGrid).toHaveBeenCalledTimes(1)
      expect(mockedDestinationList).toHaveBeenCalledTimes(1)
      expect(mockedRoutesList).toHaveBeenCalledTimes(1)
      expect(mockedAlliance).toHaveBeenCalledTimes(1)
    })
  })

  describe('Error Handling and Edge Cases', () => {
    it('handles undefined airlineData gracefully', () => {
      const { getByTestId, queryByTestId } = render(
        <AirlineFlightTab airlineData={undefined as any} />,
      )
      expect(getByTestId('stats-grid-section')).toBeTruthy()
      expect(queryByTestId('alliance-section')).toBeNull()
    })

    it('handles missing operations data', () => {
      const dataWithoutOperations: any = {
        network: mockedAirlineData.network,
      }
      const { getByTestId, queryByTestId } = render(
        <AirlineFlightTab airlineData={dataWithoutOperations} />,
      )
      expect(getByTestId('stats-grid-section')).toBeTruthy()
      expect(queryByTestId('alliance-section')).toBeNull()
    })
  })
})

describe('AirlineFlightTab Component Snapshot', () => {
  it('should render the AirlineFlightTab Component successfully', () => {
    const { toJSON } = render(<AirlineFlightTab airlineData={mockedAirlineData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
