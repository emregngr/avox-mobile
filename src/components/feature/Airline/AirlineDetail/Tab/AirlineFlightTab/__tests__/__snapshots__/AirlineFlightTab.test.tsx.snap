// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AirlineFlightTab Component Snapshot should render the AirlineFlightTab Component successfully 1`] = `
<View
  className="px-4 gap-y-4"
>
  <View
    testID="stats-grid-section"
  >
    <Text
      testID="destination-count-stat"
    >
      150
    </Text>
    <Text
      testID="destination-countries-stat"
    >
      45
    </Text>
    <Text
      testID="domestic-connections-stat"
    >
      80
    </Text>
    <Text
      testID="international-connections-stat"
    >
      70
    </Text>
    <Text
      testID="country"
    >
      country
    </Text>
    <Text
      testID="domestic"
    >
      domesticDestinations
    </Text>
    <Text
      testID="international"
    >
      internationalDestination
    </Text>
    <Text
      testID="total"
    >
      totalDestination
    </Text>
  </View>
  <View
    testID="destination-list-section"
  >
    <Text
      testID="destination-title"
    >
      mainDestinations
    </Text>
    <Text
      testID="destination-count"
    >
      3
    </Text>
  </View>
  <View
    testID="routes-list-section"
  >
    <Text
      testID="routes-title"
    >
      activeRoutes
    </Text>
    <Text
      testID="routes-count"
    >
      2
    </Text>
  </View>
  <View
    testID="alliance-section"
  >
    <Text
      testID="alliance-title"
    >
      alliance
    </Text>
    <Text
      testID="alliance-value"
    >
      Star Alliance
    </Text>
  </View>
</View>
`;
